packages/ember-metal/lib/run_loop.js

```diff
@@ -146,6 +146,9 @@ run.join = function() {
   context of the App.RichTextEditorComponent and to have the invocation of that
   method be safely handled and executed by the Ember run loop.
 
+  Please note, when the function is invoked within an existing loop, no return
```

Comment:
This has been fixed and `run.join` now properly returns a value even if inside an existing loop (was fixed with an update to Backburner before 1.9).